src/buffer/out/textBuffer.cpp

```diff
@@ -1660,6 +1660,70 @@ const std::vector<til::inclusive_rect> TextBuffer::GetTextRects(til::point start
     return textRects;
 }
 
+// Method Description:
+// - Computes the span(s) for the given selection
+// - If not a blockSelection, returns a single span (start - end)
+// - Else if a blockSelection, returns spans corresponding to each line in the block selection
+// Arguments:
+// - start: beginning of the text region of interest (inclusive)
+// - end: the other end of the text region of interest (inclusive)
+// - blockSelection: when enabled, get spans for each line covered by the block
+// - bufferCoordinates: when enabled, treat the coordinates as relative to
+//                      the buffer rather than the screen.
+// Return Value:
+// - one or more sets of start-end coordinates, representing spans of text in the buffer
+std::vector<til::point_span> TextBuffer::GetTextSpans(til::point start, til::point end, bool blockSelection, bool bufferCoordinates) const

```

Comment:
To be honest, neither of those make a ton of sense to me and I'm in the process of replacing them over time (hopefully). Given a proper text buffer implementation, there's no need to have a `_ExpandTextRow`, because the text buffer should never return cut off wide glyphs in the first place. I'm hoping that in the future you can just pass whatever selections you got without caring about `ScreenToBufferLine` or wide glyphs messing up the result, because that's IMO the only way this will ever be robust.

But if I had to vote for an API that we should keep, I'd personally vote for this new one. Given a Z pattern text selection, it returns a single selection "span", which to me makes a lot more sense than breaking the up selection into lines.